src/inausoft.netCLI/netCLI.cs

```diff
@@ -0,0 +1,99 @@
+﻿using inausoft.netCLI.Commands;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace inausoft.netCLI
+{
+    public static class netCLI
+    {
+        /// <summary>
+        /// Setups CLI flow by adding and configuring <see cref="RootCommandHandler"/>.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="SetupHelpCommand"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddCLI(this IServiceCollection services, Action<CLIConfiguration> setup)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException($"{nameof(services)} cannot be null.");
+            }
+
+            CLIConfiguration configuration = new CLIConfiguration();
+
+            setup(configuration);
+
+            services.AddSingleton(configuration);
+
+            configuration._commandMap.Values.ToList().ForEach(it =>
+            {
+                services.AddSingleton(it);
+            });
+
+            return services;
+        }
+
+        /// <summary>
+        /// Runs <see cref="ICommandHandler"/> for command specified in args./>
+        /// </summary>
+        /// <param name="serviceProvider"></param>
+        /// <param name="args">Application's command line arguments.</param>
+        /// <returns></returns>
+        public static int RunCLI(this IServiceProvider serviceProvider, string[] args)
```

Comment:
Why is ServiceProvider becoming responsible to run some CLI commands?

I'd suggest to create a non static class that would accept `IServiceProvider` in its constructor. This class could have this method `Run` or `Execute` which could be invoked with `args[]`.

Reply:
I think it only overcomplicates the solution. But I'm open for sample implementation.